Content.Server/GameTicking/Rules/BlobRuleSystem.cs

```diff
@@ -0,0 +1,183 @@
+                {
+                    if (_roundEndSystem.ExpectedCountdownEnd != null)
+                    {
+                        _roundEndSystem.CancelRoundEndCountdown(checkCooldown: false);
+                        _chatSystem.DispatchGlobalAnnouncement(Loc.GetString("blob-alert-recall-shuttle"),
+                            Loc.GetString("Station"),
+                            false,
+                            null,
+                            Color.Red);
+                    }
+                }
+
+                switch (blobRuleComp.Stage)
+                {
+                    case BlobStage.Default when comp.BlobTiles.Count < 50:
+                        continue;
+                    case BlobStage.Default:
+                        _chatSystem.DispatchGlobalAnnouncement(Loc.GetString("blob-alert-detect"),
+                            Loc.GetString("Station"),
+                            true,
+                            blobRuleComp.AlertAudio,
+                            Color.Red);
+                        blobRuleComp.Stage = BlobStage.Begin;
+                        break;
+                    case BlobStage.Begin:
+                    {
+                        if (comp.BlobTiles.Count >= 300)
+                        {
+                            _chatSystem.DispatchGlobalAnnouncement(Loc.GetString("blob-alert-critical"),
+                                Loc.GetString("Station"),
+                                true,
+                                blobRuleComp.AlertAudio,
+                                Color.Red);
+                            var stationUid = _stationSystem.GetOwningStation(ent);
+                            if (stationUid != null)
+                                _nukeCode.SendNukeCodes(stationUid.Value);
+                            blobRuleComp.Stage = BlobStage.Critical;
+                        }
+                        break;
+                    }
+                    case BlobStage.Critical:
+                    {
+                        if (comp.BlobTiles.Count >= 400)
+                        {
+                            comp.Points = 99999;
+                            _roundEndSystem.EndRound();
+                        }
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
+    private void OnRoundEndText(RoundEndTextAppendEvent ev)
+    {
+        var query = EntityQueryEnumerator<BlobRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var blob, out var gameRule))
+        {
+            if (!GameTicker.IsGameRuleAdded(uid, gameRule))
+                continue;
+
+            if (blob.Blobs.Count < 1)
+                return;
+
+            var result = Loc.GetString("blob-round-end-result", ("blobCount", blob.Blobs.Count));
+
+            // yeah this is duplicated from traitor rules lol, there needs to be a generic rewrite where it just goes through all minds with objectives
```

Comment:
Do not copypaste.

Reply:
for the future that generic rewrite was done